### ISSUE TITLE
"asm.js type error: double is not a subtype of intish" in 1.30.0

### ISSUE THREAD

**Issue by cwoffenden** [kripken/emscripten#3293]
Since moving from 1.29.12 to 1.30.0 I spotted this asm.js type error in Firefox (and the subsequent slowdown). I tracked it down to casting from `double` to either `int64_t` or `uint64_t`, reproducible with this line:

`uint64_t val = static_cast<uint64_t>(round(1.0));`

The generated code (built with -O1) errors on this line:

`$1 = +Math_abs($call) >= 1.0 ? $call > 0.0 ? (Math_min(+Math_floor($call / 4294967296.0), 4294967295.0) | 0) >>> 0 : ~~+Math_ceil(($call - +(~~$call >>> 0)) / 4294967296.0) >>> 0 : 0;`

Building with -O0 is fine, anything else errors.

**Comment by cwoffenden** [comment]
From the generated code, should the `Math_min` call be preceded by `~~+`, since the call expects to return and validate a double, from my (limited) understanding of asm.js?

**Comment by kripken** [comment]
The code from parseTools shouldn't be used on emscripten, unless fastcomp is disabled.

I can't reproduce the problem. I tried to create a testcase,

````
#include <stdint.h>
#include <math.h>

int main(void) {
  uint64_t val = static_cast<uint64_t>(round(1.0));
  return val;
}
````

and I get

````
$1 = +Math_abs($call) >= 1.0 ? $call > 0.0 ? (~~+Math_min(+Math_floor($call / 4294967296.0), 4294967295.0)) >>> 0 : ~~+Math_ceil(($call - +(~~$call >>> 0)) / 4294967296.0) >>> 0 : 0;
````

which validates ok.

Which version of firefox do you see the error in?

**Comment by cwoffenden** [comment]
It's with Firefox 36.0.4. I'm not disabling fastcomp but I thought it might be related to using the vs-tool plugin, so I tried from the command line with the same result. I turned on EMCC_DEBUG to verify that fastcomp was used (and it is) and I saw this:
````
1>GCCCOMPILE : warning root: LLVM version appears incorrect (seeing "3.4", expected "3.5")
````
I had a dig around and it looks like earlier the LLVM release was placed in `bin/Release` (as of 3.4) and now it's in `Release/bin`. With the path updated accordingly I no longer get the error.
